src/lib/walletOpener.ts

```diff
@@ -0,0 +1,40 @@
+import { attachFrame, createFrame, detatchFrame } from './frame';
+import { popupWindow } from '../lib/popup';
+import { isSafari } from './browser';
+
+export function openWalletView(url:string) {
+  let childFrame:HTMLIFrameElement|null = null;
+  let childTab:Window|null = null;
+  let messageTarget:Window|null = null;
+
+
+  if (isSafari()) {
+    childTab = popupWindow(
+      url,
+      'BLOCTO_POP',
+      640,
+      770
+    ) || null;
+    messageTarget = childTab || null;
+  } else {
+    childFrame = createFrame(url);
+    attachFrame(childFrame);
+    messageTarget = childFrame && childFrame.contentWindow;
+  }
+
+  const closeChild = () => {
```

Comment:
since the open function named `openWalletView`, I think it's better to name close function `closeWalletView` or `closeView`

Reply:
it's really a good point! [thx](https://github.com/portto/blocto-sdk/pull/29/commits/7cdf04b5cd107329a917c0e5124ae1e6535cd3fd) !